pyk/src/pyk/cterm.py

```diff
@@ -79,3 +92,6 @@ def _ml_impl(antecedents: Iterable[KInner], consequents: Iterable[KInner]) -> KI
             return consequent
 
         return mlImplies(antecedent, consequent, Sorts.GENERATED_TOP_CELL)
+
+    def with_constraint(self, new_constraint: KInner) -> 'CTerm':
+        return CTerm(mlAnd([self.config, new_constraint] + list(self.constraints), Sorts.GENERATED_TOP_CELL))
```

Comment:
I'd call it `add_constraint` instead. To me, `with_constraint` suggests "take the configuration, and couple it with the constraint provided."

```suggestion
    def add_constraint(self, constraint: KInner) -> 'CTerm':
        return CTerm(mlAnd((self.config, constraint) + self.constraints, Sorts.GENERATED_TOP_CELL))
```

Reply:
Done.